src/crypto/slow-hash.c

```diff
@@ -1767,3 +1841,129 @@ void cn_slow_hash(const void *data, size_t length, char *hash, int variant, int
 }
 
 #endif
+
+#define SEEDHASH_EPOCH_BLOCKS	2048
+#define SEEDHASH_EPOCH_LAG	64
+
+bool rx_needhash(const uint64_t height, uint64_t *seedheight) {
+  uint64_t s_height =  (height <= SEEDHASH_EPOCH_BLOCKS+SEEDHASH_EPOCH_LAG) ? 0 :
+                       (height - SEEDHASH_EPOCH_LAG - 1) & ~(SEEDHASH_EPOCH_BLOCKS-1);
+  rx_state *rx_sp;
+  rx_s_toggle = (s_height & SEEDHASH_EPOCH_BLOCKS) != 0;
+  *seedheight = s_height;
+  rx_sp = &rx_s[rx_s_toggle];
+  if (!s_height && !rx_sp->rs_cache)
+    return true;
+  return (rx_sp->rs_height != s_height);
+}
+
+typedef struct seedinfo {
+  unsigned long si_start;
+  unsigned long si_count;
+} seedinfo;
+
+static THREAD_RTYPE rx_seedthread(void *arg) {
+  seedinfo *si = arg;
+  rx_state *rx_sp = &rx_s[rx_s_toggle];
+  randomx_init_dataset(rx_dataset, rx_sp->rs_cache, si->si_start, si->si_count);
+  return NULL;
+}
+
+void rx_seedhash(const uint64_t height, const char *hash, const int miners) {
+  randomx_flags flags = RANDOMX_FLAG_DEFAULT;
+  rx_state *rx_sp = &rx_s[rx_s_toggle];
+  MUTEX_LOCK(rx_mutex);
+  if (rx_sp->rs_height == height)
+    goto leave;
+  if (use_v4_jit())
```

Comment:
Another var to make it less confusing ?

Reply:
Or just call it use_jit()? Is there any reason to jit one but not the other?

Reply:
Debugging. use_jit would be OK, yes.